nixos/tests/userborn-immutable-etc.nix

```diff
@@ -36,8 +36,8 @@ in
       specialisation.new-generation = {
         inheritParentConfig = false;
         configuration = {
-          nixpkgs = {
-            inherit (config.nixpkgs) hostPlatform;
+          nixpkgs.hostPlatform = {
+            inherit (pkgs.stdenv.hostPlatform) system;
```

Comment:
Why would non-`system` attributes have to be dropped?
This change should not be required of anyone.

Reply:
Just to be clear: I have no stakes (or particular interest) in the original PR and was just looking for a quick fix for my tests. I have no objections to reverting the original PR and this one.